src/Microsoft.Health.CosmosDb/Features/Storage/CosmosDocumentQueryFactory.cs

```diff
@@ -0,0 +1,37 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Linq;
+
+namespace Microsoft.Health.CosmosDb.Features.Storage
+{
+    /// <summary>
+    /// Factory for creating the <see cref="IDocumentQuery{T}"/>.
+    /// </summary>
+    public class CosmosDocumentQueryFactory : ICosmosDocumentQueryFactory
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CosmosDocumentQueryFactory"/> class.
+        /// </summary>
+        public CosmosDocumentQueryFactory()
+        {
+        }
+
+        /// <inheritdoc />
+        public IDocumentQuery<T> Create<T>(IDocumentClient documentClient, CosmosQueryContext context)
+        {
+            EnsureArg.IsNotNull(documentClient, nameof(documentClient));
+            EnsureArg.IsNotNull(context, nameof(context));
+
+            return documentClient.CreateDocumentQuery<T>(
```

Comment:
We probably want to move the logging stuff into here so the query/RUs can be logged. Doesn't have to be in this PR. Maybe we can do it together when we refactor the code generator.

Reply:
Took me a sec to realize why this class was here